hibernate-core/src/main/java/org/hibernate/bytecode/internal/BytecodeProviderInitiator.java

```diff
@@ -43,12 +85,15 @@ public Class<BytecodeProvider> getServiceInitiated() {
 
 	@Internal
 	public static BytecodeProvider buildDefaultBytecodeProvider() {
-		return buildBytecodeProvider( BYTECODE_PROVIDER_NAME_BYTEBUDDY );
+		return getBytecodeProvider( ServiceLoader.load( BytecodeProvider.class ) );
 	}
 
+	/**
+	 * @deprecated Register a {@link BytecodeProvider} through Java {@linkplain java.util.ServiceLoader services}.
+	 */
 	@Internal
+	@Deprecated( forRemoval = true )
 	public static BytecodeProvider buildBytecodeProvider(String providerName) {
-
```

Comment:
I like that you're being conservative, but considering it's marked as internal it might be ok to just remove this method, or perhaps replace its body with a "throw error".
My reasoning is that we should get rid of those references to the `org.hibernate.bytecode.internal.bytebuddy.BytecodeProviderImpl` as soon as possible..

But I agree perhaps do it in a separate commit, so to change as least as possible for the patch to be backported, for example I'm not sure what impact it would have on Spring native as they also have some needs in this area.

Reply:
Added a separate commit that removes the method.